MEPTS-1243 | MQ CAT 15 - Changes for June 2023

iteliolucas opened the pull request.

iteliolucas committed:
Updates transferred out query for MQ Cat 15

iteliolucas committed:
Updates dead patients composition to remove Death Registered in Last Home Visit Card

iteliolucas committed:
Adds new DA criteria to Utentes Inscritos no MDS  há 24 meses - Adultos

iteliolucas committed:
Adds new DA criteria to Tipo de dispensa

iteliolucas committed:
Adds new criteria to próximo levantamento FILA

iteliolucas committed:
Adds new DA criteria to utentes com dois pedidos de CV

iteliolucas committed:
Removes criterias for Data da Última Consulta 3º Mês, 2º Mês and 1º Mês

iteliolucas committed:
Changes on Num 15.13 Oferta/Início MDS

iteliolucas committed:
Updates MDS suspension to check next appointment instead of next pickup

iteliolucas committed:
Removes exclusion criteria from Den 15.15

iteliolucas committed:
Adds RF45 Utentes que estão em tratamento de TB (DSD_FR10)

iteliolucas committed:
Adds RF47: Utentes que tiveram alguma reacção a medição nos últimos 6m

iteliolucas committed:
Adds RF46: Utentes que tiveram alguma vez Sarcoma de Kaposi (SK)

iteliolucas committed:
Adds RF50: Utentes que tiveram alguma vez Sarcoma de Kaposi (SK)

iteliolucas committed:
Merge branch '2.x_develop' into MEPTS-1243-MQ_CAT_15_Changes_for_June_2023

eliazmuchave commented on api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java:
```diff
@@ -12307,4 +12369,111 @@ public CohortDefinition getPatientsWithCargaViralonFichaClinicaAndFichaResumo(in
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("9", hivMetadata.getPediatriaSeguimentoEncounterType().getEncounterTypeId());
+    map.put("507", hivMetadata.getKaposiSarcomaConcept().getConceptId());
+    map.put("1406", hivMetadata.getOtherDiagnosis().getConceptId());
+
+    String query =
+        " SELECT p.patient_id "
+            + "        FROM patient p "
+            + "             INNER JOIN encounter e "
+            + "                ON p.patient_id = e.patient_id "
+            + "             INNER JOIN obs o "
+            + "                ON e.encounter_id = o.encounter_id "
+            + "        WHERE p.voided = 0 "
+            + "             AND e.voided = 0 "
+            + "             AND o.voided = 0 "
+            + "             AND e.encounter_type IN (${6},${9}) "
+            + "             AND o.concept_id = ${1406} "
+            + "             AND o.value_coded = ${507} "
+            + "             AND e.location_id = :location "
```
Group by patient Id

eliazmuchave commented on api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java:
```diff
@@ -8636,14 +8687,14 @@ public CohortDefinition getMQ15DenMDS() {
     CohortDefinition Mq15F = intensiveMonitoringCohortQueries.getMI15F();
     CohortDefinition Mq15G = intensiveMonitoringCohortQueries.getMI15G();
     CohortDefinition alreadyMds = getPatientsAlreadyEnrolledInTheMdc();
+    CohortDefinition onTB = commonCohortQueries.getPatientsOnTbTreatment();
```
It seems to me that the part of the query that implements "Condições Médicas Importantes" is not in compliance with the requirement. Please review the How to Compute algorithm

iteliolucas committed:
Adds group by to Patients With Sarcoma Karposi query

iteliolucas committed:
Fixes other diagnosis mapping

iteliolucas committed:
Merge branch '2.x_develop' into MEPTS-1243-MQ_CAT_15_Changes_for_June_2023

iteliolucas committed:
Adds git review recomendations

1. Removes Max function under encounter_datetime and obs_datetime as recommended

2. Adds GROUP BY to the union queries as recommended

iteliolucas committed:
MVN clean install

iteliolucas committed:
Reverts Patients On Tb Treatment query to get last treatment plan

eliazmuchave reviewed (approved):
Looks good to me

iteliolucas committed:
Merge branch '2.x_develop' into MEPTS-1243-MQ_CAT_15_Changes_for_June_2023

iteliolucas committed:
Removes age cohort duplicate
